Beaker

greg-hellings opened the pull request:
A first attempt at syncrhonous-only Beaker provisioning.

herlo commented:
Please also rebase against the master branch to obtain the latest requirements.txt

herlo commented:
bump @greg-hellings. Any progress on this?

greg-hellings commented:
@herlo Yes, sorry I've been away from dev for the past 2 weeks - work travel for a week, then vacation for a week. I have addressed most of the issues, I'm just looking to finish up the tear-down implementation.

greg-hellings commented on ex_schemas/schema_v2.json:
```diff
@@ -50,7 +50,8 @@
                                    {"$ref": "#/definitions/ex_res"},
```
I can't make heads or tails out of schema_v3, sorry.

greg-hellings commented:
@herlo I've merged master and added tear-down.

greg-hellings commented:
OK, that should cover the schema_v3 changes.

herlo commented:
centos-paas-test

arilivigni commented:
centos-paas-test

arilivigni commented:
centos-paas-test

arilivigni commented:
centos-paas-test

arilivigni commented:
centos-paas-test

greg-hellings commented:
Is  this going to be merged?

herlo commented:
@greg-hellings yes, that is the plan. I'm in training this week and haven't had time to test the teardown. I'll be doing that on Monday. Sorry for the delay and noise.

herlo commented:
After quite a review of the code, I'm fairly concerned with the way that teardown is happening. The concept is that the output should be used to query the server to be able to teardown the systems without modifying the topology. 

@greg-hellings, you mentioned that there is no good documentation for this process, which is true, as we're breaking new ground on provisiong/teardown. However, there is a model that has been used in every other plugin currently written in linch-pin. 

The timeline is tight, but @samvarankashyap is willing to work with you in the next couple of days to try to rework the beaker module to ensure it can be merged and follow the model of the other plugins. Would that be something you would be willing to do right away??

If not, I can continue down the path of reworking the module, but I'm definitely not as familiar with beaker, and it may delay any plans you have with a presentation. Alternatively, you could just use this fork/branch for any demonstrations.

Thoughts?

greg-hellings commented:
I don't have time allocated in the current sprint to continue chasing a set of hazily moving goal posts with this. If you want drastic changes like that from me, I'll schedule them into a sprint once the procedure and features needed are clearly documented. Otherwise, you can do as you wish with this branch.
